### ISSUE TITLE
[BUG] NullPointerException when setting the text of shop sign

### ISSUE THREAD

**Issue by wattywatty14** [Ghost-chu/QuickShop-Reremake#777]
**Describe the bug**
Players can't buy items from ChestShops.

**To Reproduce**
1. Go to 'Download the latest version of the plugin.'
2. Have a player attempt to buy an item.

**Expected behavior**
I expected players to be able to buy items. 

**Screenshots**

**Paste link:**
 Execute command /qs paste, you will get a link contains your server information, paste it under this text.
 You must create a paste, except plugin completely won't work.
 If you create failed, you should find a paste file under the plugin/QuickShop folder.
- 【 https://paste.ubuntu.com/p/hhFDpNfF7k/ 】

**Additional context**
Console error: https://paste.ubuntu.com/p/GbrfW2tSp8/

**Comment by Ghost-chu** [comment]
update

**Comment by sandtechnology** [comment]
It shouldn't happen because we have the null checking, so look like is server software issue, how about rollback the paper version?

**Comment by wattywatty14** [comment]
@sandtechnology I'd rather not rollback the paper version, it patches a dupe exploit. 

@Ghost-chu We have the latest Paper-Spigot and the latest version of the plugin.

**Comment by sandtechnology** [comment]
But we have check if is null already, and the exception was thrown from bukkit side, nothing we can do there.
Please report it to paper if this version still doesn't work: https://ci.codemc.io/job/Ghost-chu/job/QuickShop-Reremake-SNAPSHOT/479/

**Comment by Ghost-chu** [comment]
no support for outdated version

**Comment by Ghost-chu** [comment]
I'm sorry that I just saw the version is 4.0.6.1 not 4.0.3.1

**Comment by wattywatty14** [comment]
@sandtechnology @Ghost-chu Still getting this error:

```
[05:54:02 ERROR]: Could not pass event PlayerInteractEvent to QuickShop v4.0.6.1
java.lang.NullPointerException: null
        at org.bukkit.craftbukkit.v1_16_R3.block.CraftSign.applyTo(CraftSign.java:79) ~[patched_1.16.4.jar:git-Paper-356]
        at org.bukkit.craftbukkit.v1_16_R3.block.CraftSign.applyTo(CraftSign.java:13) ~[patched_1.16.4.jar:git-Paper-356]
        at org.bukkit.craftbukkit.v1_16_R3.block.CraftBlockEntityState.update(CraftBlockEntityState.java:147) ~[patched_1.16.4.jar:git-Paper-356]
        at org.bukkit.craftbukkit.v1_16_R3.block.CraftBlockState.update(CraftBlockState.java:167) ~[patched_1.16.4.jar:git-Paper-356]
        at org.maxgamer.quickshop.shop.ContainerShop.setSignText(ContainerShop.java:884) ~[?:?]
        at org.maxgamer.quickshop.shop.ContainerShop.setSignText(ContainerShop.java:622) ~[?:?]
        at org.maxgamer.quickshop.shop.ContainerShop.onClick(ContainerShop.java:375) ~[?:?]
        at org.maxgamer.quickshop.listener.PlayerListener.onClick(PlayerListener.java:123) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor888.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.4.jar:git-Paper-356]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.4.jar:git-Paper-356]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.4.jar:git-Paper-356]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.4.jar:git-Paper-356]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:505) ~[patched_1.16.4.jar:git-Paper-356]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:472) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.PlayerInteractManager.a(PlayerInteractManager.java:154) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:1485) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.PacketPlayInBlockDig.a(SourceFile:40) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.PacketPlayInBlockDig.a(SourceFile:10) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:23) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1133) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1126) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:1087) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1001) ~[patched_1.16.4.jar:git-Paper-356]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:178) ~[patched_1.16.4.jar:git-Paper-356]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
```

**Comment by sandtechnology** [comment]
Please report it to paper, nothing we can do there.

**Comment by sandtechnology** [comment]
Have been fixed by  Paper#4976
